src/formatDate.js

```diff
@@ -50,7 +50,45 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const [ a, b, c, separator ] = fromFormat;
+  const [ , , , newSeparator ] = toFormat;
+
+  const newData = [];
+
+  const arr = date.split(separator);
+
+  const obj = {};
+
+  obj[a] = arr[0];
+  obj[b] = arr[1];
+  obj[c] = arr[2];
+
+  if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
+    obj.YYYY = (obj.YYYY).substring(2);
+  }
+
+  if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    if (obj.YY < 30) {
+      obj.YYYY = `20${obj.YY}`;
+    } else {
+      obj.YYYY = `19${obj.YY}`;
+    }
+  }
```

Comment:
```suggestion
  if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
    obj.YYYY = (obj.YYYY).substring(2);
    
    if (obj.YY < 30) {
      obj.YYYY = `20${obj.YY}`;
    } else {
      obj.YYYY = `19${obj.YY}`;
    }
  }
```

Reply:
With these changing tests are faild
      ✕ if we shorten YYYY to YY (4ms)
      ✕ if we extend YY < 30 to YYYY (31ms)
      ✕ if we extend YY = 00 to YYYY (2ms)
      ✕ if we extend YY = 30 to YYYY
      ✕ if we extend YY > 30 to YYYY (2ms)